texar/core/optimization.py

```diff
@@ -271,12 +272,17 @@ def get_grad_clip_fn(hparams: Optional[Union[HParams,
     return grad_clip_fn
 
 
-def get_train_op(optimizer: Optimizer,
+def get_train_op(params: Optional[Union[List[torch.Tensor],
```

Comment:
[Minor issue] The type of `params` looks very complex. Is it necessary or can it be simplified? (ok if not)